src/wayland-eglhandle.c

```diff
@@ -151,6 +152,15 @@ wlEglCreatePlatformData(int apiMajor, int apiMinor, const EGLExtDriver *driver)
     res->callbacks.setError           = driver->setError;
     res->callbacks.streamSwapInterval = driver->streamSwapInterval;
 
+    /* try to find libdrm and drmGetDeviceFromDevId */
+    res->libdrmDlHandle = dlopen("libdrm.so.2", RTLD_LAZY);
```

Comment:
Minor nitpick, but it might be better to try RTLD_DEFAULT first, and then go with dlopen if that fails.

In most cases, the RTLD_DEFAULT case will succeed, so we'd avoid the extra overhead of calling dlsym.

Also, using RTLD_DEFAULT first means that you could still override `drmGetDeviceFromDevId` with an LD_PRELOAD library if you had some reason to do so.

Either way, we should add a comment here explaining why loading libdrm.so can give a different result than using RTLD_DEFAULT, since by the spec the two should be strictly equivalent.